### DISCUSSION TITLE
Reusable workflows internal repositories feature removal?

### DISCUSSION THREAD

**Discussion by Zejnilovic** [community/community#25297] [Enterprise]
<p>Hello,</p>
<p>I noticed 2 docs have a bit of different info. One is saying reusable workflows work with internal repositories and the second one doesn’t.</p>
<p>1st link <a href="https://docs.github.com/en/actions/learn-github-actions/reusing-workflows" class="inline-onebox" rel="noopener nofollow ugc">Reusing workflows - GitHub Docs</a></p>
<pre><code class="lang-auto">A reusable workflow can be used by another workflow if either of the following is true:

- Both workflows are in the same repository.
- The called workflow is stored in a public repository.
</code></pre>
<p>2nd link <a href="https://docs.github.com/en/enterprise-cloud@latest/actions/learn-github-actions/reusing-workflows" class="inline-onebox" rel="noopener nofollow ugc">Reusing workflows - GitHub Docs</a></p>
<pre><code class="lang-auto">A reusable workflow can be used by another workflow if any of the following is true:

- Both workflows are in the same repository.
- The called workflow is stored in a public repository.
- The called workflow is stored in an internal repository and the settings for that repository allow it to be accessed. For more information, see "Managing GitHub Actions settings for a repository."
</code></pre>
<p>Anyone who could elaborate on this? I just spent 2 hours setting this up in our org and it seems it was for nought. I get:</p>
<pre><code class="lang-vim">error parsing called workflow "org/reusable-workflows/.github/workflows/tf_wf.yml@develop": 
Workflows in 'org/my-repo' cannot access remote workflows in 'org/reusable-workflows'. 
See https://docs.github.com/en/actions/learn-github-actions/reusing-workflows#access-to-reusable-workflows for more information.
</code></pre>
<p>Thanks,<br>
Sasa</p>

**Comment by Katsute** [comment]
<p>It seems that reusable workflows for internal repositories are exclusively for enterprise users <a href="https://docs.github.com/en/enterprise-cloud@latest/actions/learn-github-actions/reusing-workflows#access-to-reusable-workflows" class="inline-onebox" rel="noopener nofollow ugc">Reusing workflows - GitHub Docs</a></p>

**Comment by jshier** [comment]
<p>This seems incorrect. The document you link to has two points in regards to access:</p>
<blockquote>
<ul>
<li>The called workflow is stored in a public repository.</li>
<li>The called workflow is stored in an internal repository and the settings for that repository allow it to be accessed. For more information, see “<a href="https://docs.github.com/en/enterprise-cloud@latest/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository#allowing-access-to-components-in-an-internal-repository" rel="noopener nofollow ugc">Managing GitHub Actions settings for a repository</a>.”</li>
</ul>
</blockquote>
<p>By my reading, private repos should work as long as the workflow’s repo is accessible.</p>

**Comment by laughedelic** [comment]
<p>I think it’s a terminology confusion: internal and private are not the same. A repository can be</p>
<ul>
<li>public: visible to anyone</li>
<li>private: visible only to the owner (e.g. org admins) and collaborators</li>
<li>internal: visible to anyone in the org (enterprise account)</li>
</ul>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://github.blog/2019-11-05-internal-repositories-are-now-generally-available-for-github-enterprise/">
  <header class="source">
      <img src="https://github.blog/wp-content/uploads/2019/01/cropped-github-favicon-512.png?fit=32%2C32" class="site-icon" width="32" height="32">

      <a href="https://github.blog/2019-11-05-internal-repositories-are-now-generally-available-for-github-enterprise/" target="_blank" rel="noopener nofollow ugc" title="05:00PM - 05 November 2019">The GitHub Blog – 5 Nov 19</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/361;"><img src="https://user-images.githubusercontent.com/766656/181090135-d13e9e53-3764-45d4-86b5-31c07fd2b652.png" class="thumbnail" width="690" height="361" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/c/0/c017c230d6e8ab7ae3a76b88d5272b38b442726f_2_690x361.png, https://aws1.discourse-cdn.com/github/optimized/3X/c/0/c017c230d6e8ab7ae3a76b88d5272b38b442726f_2_1035x541.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/c/0/c017c230d6e8ab7ae3a76b88d5272b38b442726f.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/c/0/c017c230d6e8ab7ae3a76b88d5272b38b442726f_2_10x10.png"></div>

<h3><a href="https://github.blog/2019-11-05-internal-repositories-are-now-generally-available-for-github-enterprise/" target="_blank" rel="noopener nofollow ugc">Internal repositories are now generally available for GitHub Enterprise</a></h3>

  <p>Now out of beta, the internal repository visibility allows an enterprise-owned repository to be read by any member of any organization that belongs to an enterprise account.</p>

  <p>
    <span class="label1">Est. reading time: 3 minutes</span>
  </p>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<aside class="onebox allowlistedgeneric" data-onebox-src="https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/setting-repository-visibility#making-a-repository-internal">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">

      <a href="https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/setting-repository-visibility#making-a-repository-internal" target="_blank" rel="noopener nofollow ugc">GitHub Docs</a>
  </header>

  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/766656/181090136-9d78c4e9-62ad-46f6-9ce9-a0c5defc26cb.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_10x10.png">

<h3><a href="https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/setting-repository-visibility#making-a-repository-internal" target="_blank" rel="noopener nofollow ugc">Setting repository visibility - GitHub Docs</a></h3>



  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

**Comment by tgrall** [comment, marked as answer]
<p>Hello,</p>
<p>So since you are on GitHub Enterprise Cloud you can put a reusable worflow in a <strong>internal</strong> repository, and use it in other repositories of you organization and/or enterprise, but you must give access/permissions.</p>
<ol>
<li>Go to you Reusable Workflow repository (the one that contains the workflow you want to call)</li>
<li>Click on <strong>Settings</strong>
</li>
<li>Click on <strong>Actions</strong>
</li>
<li>Go at the bottom of the configuration page, in the <strong>Access</strong> section</li>
<li>Then select the proper option<br>
<div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/1/9/192ba319bfa26d52962da455ac69e4599637b3cd.png" data-download-href="/uploads/short-url/3AFqAsZCoyn09unnb1F8sEg0hHv.png?dl=1" title="Screenshot 2021-12-02 at 11.17.51" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/541250/181090137-636f6c8d-a6b7-448a-b199-f24f8b6ba685.png" alt="Screenshot 2021-12-02 at 11.17.51" data-base62-sha1="3AFqAsZCoyn09unnb1F8sEg0hHv" width="690" height="259" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/1/9/192ba319bfa26d52962da455ac69e4599637b3cd_2_690x259.png, https://aws1.discourse-cdn.com/github/original/3X/1/9/192ba319bfa26d52962da455ac69e4599637b3cd.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/1/9/192ba319bfa26d52962da455ac69e4599637b3cd.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/1/9/192ba319bfa26d52962da455ac69e4599637b3cd_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">Screenshot 2021-12-02 at 11.17.51</span><span class="informations">878×330 37.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
</li>
</ol>
<p>This should now allow your “caller” repository to call the reusable workflow.</p>
<p>Regards<br>
<a href="https://twitter.com/tgrall" rel="noopener nofollow ugc">Tug</a></p>

**Comment by MageshSrinivasulu** [comment]
<p><a class="mention" href="/tgrall">@tgrall</a> I am not using enterprise repo but I am able to create internal repo in the organization. I gave the appropriate permission for the internal repo but, still not able to access it from the calling repo. Is this expected?</p>

**Comment by dataviruset** [comment]
With this change, does it mean we don't need to be on GitHub Enterprise anymore in order to share workflows between repositories? https://github.blog/changelog/2022-12-14-github-actions-sharing-actions-and-reusable-workflows-from-private-repositories-is-now-ga/

**Reply by ChristopherHX** [reply]
Yes I tested it and it works with GitHub Free if both caller and reusable workflows are in different private repos of the same Account and you enable the new repo setting.

**Comment by MikeDabrowski** [comment]
Something is wrong with the permissions here. We were using shared workflows in many repos, all of them private. And one repo was somehow created as an internal - and this one couldn't access them!

**Reply by k-patrik** [reply]
Same observation here! After converting the calling repository from `internal` into `private` the reusable workflow worked.

**Reply by cvette** [reply]
Can confirm, this does not work with internal repositories right now.
